### PR TITLE
Wagtail: Add aria-label option to links

### PR DESCRIPTION
**Pull request by anselmbradford**
Pages like https://www.consumerfinance.gov/owning-a-home/ have links like "Learn more" that are not descriptive enough. This PR adds the ability to add an aria-label to links in Wagtail that aren't descriptive enough for assistive devices.

## Additions

- Adds aria-label input field to Wagtail hyperlinks.

## How to test this PR

1. Visit a page with a hyperlink like Buying a House (http://localhost:8000/admin/pages/3701/edit/) and see that you can add an aria label to link.

## Screenshots

Screen Shot 2021-06-22 at 1 05 49 PM

### CODE REVIEW

File: cfgov/v1/atomic_elements/atoms.py [hunk 1]
```diff
@@ -26,6 +26,14 @@ class Meta:
...
 class Hyperlink(blocks.StructBlock):
     text = blocks.CharBlock(required=False)
+    aria_label = blocks.CharBlock(required=False,
+                                  help_text='Add an ARIA label if the link '
+                                            'text does not describe the '
+                                            'destination of the link '
+                                            '(e.g. has ambiguous text like '
+                                            '"Learn more" that is not '
+                                            'descriptive on its own).'
+    )
```
Comment (willbarton) [inline cfgov/v1/atomic_elements/atoms.py:36]: Minor formatting suggestion:

```suggestion
    aria_label = blocks.CharBlock(
        required=False,
        help_text=(
            'Add an ARIA label if the link text does not describe the '
            'destination of the link (e.g. has ambiguous text like '
            '"Learn more" that is not descriptive on its own).'
        )
    )
```

File: cfgov/v1/atomic_elements/organisms.py [hunk 2]
```diff
@@ -1061,9 +1062,10 @@ def links(self):
         for hyperlink in self.get('links') or []:
             url = hyperlink.get('url')
             text = hyperlink.get('text')
+            aria_label = hyperlink.get('aria_label')
 
             if url and text:
-                links.append({'url': url, 'text': text})
+                links.append({'url': url, 'text': text, 'aria_label': aria_label})
```
Comment (willbarton) [inline cfgov/v1/atomic_elements/organisms.py:1068]: This should satisfy the linter as well:

```suggestion
                links.append({
                    'url': url, 
                    'text': text, 
                    'aria_label': aria_label
                })
```

**Review by willbarton — APPROVED** [review]
Python looks good to me.

File: cfgov/v1/tests/atomic_elements/organisms/test_organisms.py [hunk 3]
```diff
@@ -445,7 +445,8 @@ def test_render(self):
             ('/bar/', 'Another link'),
         ):
             self.assertIn(
-                f'<a class="m-list_link" href="{url}">{text}</a>',
+                f'<a class="m-list_link"\n                       '
+                f'href="{url}"\n                       >{text}</a>',
```
Comment (anselmbradford) [inline cfgov/v1/tests/atomic_elements/organisms/test_organisms.py:449]: Is there a better way to do this @willbarton ?

**Comment by Scotchester** [comment]
Looks like this accomplished most of what I was intending to eventually get around to on #4096, so that can probably be closed.

**Comment by anselmbradford** [comment]
> Looks like this accomplished most of what I was intending to eventually get around to on #4096, so that can probably be closed.

Oh, look at that! I forgot about the overlap there and it indeed looks very similar… which makes me question why we didn't merge yours and if I'm missing an issue in mine?

**Comment by Scotchester** [comment]
Mine had only added it to the Hyperlink atom template, and I realized there places where the Hyperlink atom was used on the back end, but when it was rendered, the Hyperlink template wasn't used. One of those was the FCM, as you updated in this PR. I have a suspicion there are others, as well.

One place that's a little more complicated but definitely needs `aria-label`s is ResourceLists, where there usually ends up being a whole column of `Download` or `Order` links. Those should probably just be handled in the template automatically (i.e., don't create a place to manually enter an `aria-label` for each of those, but rather automatically add on to the specific link text, like `arial-label="{{ link text }} {{ resource name }}"` (pseudocode).

**Pull request merged**